### PR TITLE
Isolate Hardhat Network Explanation & add a Reference

### PR DESCRIPTION
**Pull request by feuGeneA**
- [x] Convert existing document to be purely an Explanation, moving all Reference details to a separate Reference document.
- [x] Refine and expand the new reference document.
- [x] Update the website layout to include these new documents. Done in https://github.com/nomiclabs/hardhat/pull/1657/commits/d247763ee2afee0abdbf7d07ee98506e8a4df672
- [x] Ensure no links are broken, and add additional inter-links between the Explanation and the Reference. Done in https://github.com/nomiclabs/hardhat/pull/1657/commits/ce33c0d8ff1fefbf4c1e92fb3be8d503204c01b3
- [x] Say just "Hardhat," never "Hardhat Runner," to avoid confusing people.  Comment out "Runner", with a TODO in the comment. Done in https://github.com/nomiclabs/hardhat/pull/1657/commits/82d4a460cd305e5d9e8753821bf019af2611f809
- [x] After July 21 release, rebase this PR on the `website` branch.

### CODE REVIEW

**Comment by alcuadrado** [comment]
I think this now conflicts with the changes I made in this PR: https://github.com/nomiclabs/hardhat/pull/1705

File: docs/hardhat-network/README.md [hunk 1]
```diff
@@ -1,41 +1,44 @@
-# Hardhat Network
+# Hardhat Network: An Explanation <!--theoretical, not practical; for studying, not for working-->
```
Comment (alcuadrado) [inline docs/hardhat-network/README.md:1]: Do we need the `: An Explanation` part? I would assume that this section explains what Hardhat network is.

File: docs/hardhat-network/README.md [hunk 2]
```diff
@@ -1,41 +1,44 @@
...
-- It's backed by the `@ethereumjs/vm` EVM implementation, the same one used by ganache, Remix and Ethereum Studio.
-- It supports the following hardforks:
-  - byzantium
-  - constantinople
-  - petersburg
-  - istanbul
-  - muirGlacier
+It runs as either an in-process or stand-alone daemon, servicing JSON-RPC and WebSocket requests.
+
+It mines a block with each transaction that it receives, in order and with no delay.
```
Comment (alcuadrado) [inline docs/hardhat-network/README.md:11]: This is now the default, but can be changed, as explained here https://hardhat.org/hardhat-network/#mining-modes

File: docs/hardhat-network/README.md [hunk 3]
```diff
@@ -269,162 +147,18 @@ eth_call
...
-const trace = await hre.network.provider.send("debug_traceTransaction", [
-  "0x123...",
-  {
-    disableMemory: true,
-    disableStack: true,
-    disableStorage: true,
-  },
-]);
-```
+## Why _wouldn't_ I want to use it?
```
Comment (alcuadrado) [inline docs/hardhat-network/README.md:158]: I'm not sure if I agree with this subtitle. IMO the limitation section was mostly to help users understand what can go wrong and how to deal with those (e.g. disable the optimizer).

Maybe this should be moved to reference?

File: docs/hardhat-network/README.md [hunk 4]
```diff
@@ -82,164 +85,39 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
...
-
-### Configuring mining modes using RPC methods
-
-You can change the mining behavior on runtime using two RPC methods: `evm_setAutomine` and `evm_setIntervalMining`. For example, to disable automining:
+You can use one of these modes, both or neither. By default, only the automine mode is enabled.
 
-```js
-await network.provider.send("evm_setAutomine", [false]);
-```
+When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that transactions are prioritized by gas price, then arrival time, and that invalid transactions are dropped. Pending transactions can be queried via the `eth_getBlockByNumber` RPC method (with `"pending"` as the block number argument), they can be removed using the `hardhat_dropTransaction` RPC method, and they can be replaced by a new transaction with the same nonce but a gas price 10+% higher.
```
Comment (alcuadrado) [inline docs/hardhat-network/README.md:114]:
```suggestion
When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that transactions are prioritized by gas/fee prices, then arrival time, and that invalid transactions are dropped. Pending transactions can be queried via the `eth_getBlockByNumber` RPC method (with `"pending"` as the block number argument), they can be removed using the `hardhat_dropTransaction` RPC method, and they can be replaced by a new transaction with the same nonce but a gas price 10+% higher.
```

London changed this

File: docs/hardhat-network/README.md [hunk 5]
```diff
@@ -82,164 +85,39 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
...
-      auto: false,
-      interval: [3000, 6000]
-    }
-  }
-}
-```
+To use it, you simply import `hardhat/console.sol` and call it. It implements the same formatting options that can be found in Node.js' [`console.log`](https://nodejs.org/dist/latest-v12.x/docs/api/console.html#console_console_log_data_args), which in turn uses [`util.format`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_format_format_args). For example: `console.log("Changing owner from %s to %s", currentOwner, newOwner)`.
 
-In this case, a new block will be mined after a random delay of between 3 and 6 seconds. For example, the first block could be mined after 4 seconds, the second block 5.5 seconds after that, and so on.
+It always works, regardless of the call or transaction failing or being successful, and, because it's implemented in standard Solidity, it works with any other tool or library: Remix, Waffle, Truffle, ethers.js, web3.js, etc.
```
Comment (alcuadrado) [inline docs/hardhat-network/README.md:94]: This can be somewhat confusing, as it only works in Hardhat Network and Tenderly. In other tools it's just a no-op.

File: docs/hardhat-network/reference/README.md [hunk 6]
```diff
@@ -0,0 +1,407 @@
+# Hardhat Network Reference
+
+## Supported hardforks
+
+- byzantium
+- constantinople
+- petersburg
+- istanbul
+- muirGlacier
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:9]:
```suggestion
- muirGlacier
- london
```

File: docs/hardhat-network/reference/README.md [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+
+## Config
+
+- Default behavior: Hardhat will automatically run in-process when `defaultNetwork` is empty or set to `hardhat`.
+
+### Mining modes
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:15]: Is this intended to bu a subsection of Config?

Comment (alcuadrado) [reply docs/hardhat-network/reference/README.md:15]: Well, this section is about config. I think what confuses me are the following ones, that mostly use the RPC

File: docs/hardhat-network/reference/README.md [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+}
+```
+
+This means that no new blocks will be mined by the Hardhat Network, but you can manually mine new blocks using the `evm_mine` RPC method. This will generate a new block that will include as many pending transactions as possible.
+
+### Mempool behavior
+
+When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that:
+
+- Transactions with a higher gas price are included first
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:68]:
```suggestion
- Transactions with a higher gas/fees prices are included first
```

File: docs/hardhat-network/reference/README.md [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+### Removing and replacing transactions
+
+Transactions in the mempool can be removed using the `hardhat_dropTransaction` method:
+
+```js
+const txHash = "0xabc...";
+await network.provider.send("hardhat_dropTransaction", [txHash]);
+```
+
+You can also replace a transaction by sending a new one with the same nonce as the one that it's already in the mempool but with a higher gas price. Keep in mind that, like in Geth, for this to work the new gas price has to be at least 10% higher than the gas price of the current transaction.
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:90]:
```suggestion
You can also replace a transaction by sending a new one with the same nonce as the one that it's already in the mempool but with a higher gas price. Keep in mind that, like in Geth, for this to work the new gas/fees prices has to be at least 10% higher than the gas price of the current transaction.
```

File: docs/hardhat-network/reference/README.md [hunk 10]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `hardhat_addCompilationResult`
+
+Add information about compiled contracts
+
+#### `hardhat_dropTransaction`
+
+Remove a transaction from the mempool
+
+#### `hardhat_getStackTraceFailuresCount`
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:257]: I think I wouldn't document this. It's mostly an internal API.

File: docs/hardhat-network/reference/README.md [hunk 11]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `hardhat_impersonateAccount`
+
+see the [Mainnet Forking guide](../../guides/mainnet-forking.md)
+
+#### `hardhat_intervalMine`
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:263]: Same. Also internal.

File: docs/hardhat-network/reference/README.md [hunk 12]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `hardhat_dropTransaction`
+
+Remove a transaction from the mempool
+
+#### `hardhat_getStackTraceFailuresCount`
+
+#### `hardhat_impersonateAccount`
+
+see the [Mainnet Forking guide](../../guides/mainnet-forking.md)
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:261]:
```suggestion
See the [Mainnet Forking guide](../../guides/mainnet-forking.md)
```

File: docs/hardhat-network/reference/README.md [hunk 13]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `hardhat_intervalMine`
+
+#### `hardhat_reset`
+
+see the [Mainnet Forking guide](../../guides/mainnet-forking.md)
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:267]:
```suggestion
See the [Mainnet Forking guide](../../guides/mainnet-forking.md)
```

File: docs/hardhat-network/reference/README.md [hunk 14]
```diff
@@ -0,0 +1,407 @@
...
+
+Modifies the code of an account.
+
+#### `hardhat_setLoggingEnabled`
+
+Enable or disable logging in Hardhat Network
+
+#### `hardhat_setMinGasPrice`
+
+change the minimum gas price accepted by the network (in wei)
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:283]:
```suggestion
Change the minimum gas price accepted by the network (in wei)
```

Comment (alcuadrado) [reply docs/hardhat-network/reference/README.md:283]: This method doesn't work when running London

File: docs/hardhat-network/reference/README.md [hunk 15]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `hardhat_setNonce`
+
+Modifies an account's nonce by overwriting it. Throws an InvalidInputError if nonce is smaller than the current one. The reason for this restriction is to avoid collisions when deploying contracts using the same nonce more than once.
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:287]:
```suggestion
Modifies an account's nonce by overwriting it. Throws an `InvalidInputError` if nonce is smaller than the current one. The reason for this restriction is to avoid collisions when deploying contracts using the same nonce more than once.
```

File: docs/hardhat-network/reference/README.md [hunk 16]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `hardhat_setStorageAt`
+
+Writes a single position of an account's storage. The storage position index must not exceed 2^256, and the value to write must be exactly 32 bytes long.
+
+#### `hardhat_stopImpersonatingAccount`
+
+see the [Mainnet Forking guide](../../guides/mainnet-forking.md)
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:295]:
```suggestion
See the [Mainnet Forking guide](../../guides/mainnet-forking.md)
```

File: docs/hardhat-network/reference/README.md (hunk 12 above)
Comment (alcuadrado) [reply docs/hardhat-network/reference/README.md:261]: Note: This also works when you are not forking a remote network.

File: docs/hardhat-network/reference/README.md [hunk 17]
```diff
@@ -0,0 +1,407 @@
...
+
+### Special testing/debugging methods
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:297]: I consider `debug_traceTransaction` a standard method.

File: docs/hardhat-network/reference/README.md [hunk 18]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+##### Known limitations
+
+- You can't trace transactions that use a hardfork older than [Spurious Dragon](https://ethereum.org/en/history/#spurious-dragon)
+- The last step of a message is not guaranteed to have a correct value in the `gasCost` property
+
+#### `evm_increaseTime`
+
+same as Ganache.
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:331]:
```suggestion
Same as Ganache.
```

File: docs/hardhat-network/reference/README.md [hunk 19]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `evm_mine`
+
+same as Ganache
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:335]:
```suggestion
Same as Ganache
```

File: docs/hardhat-network/reference/README.md [hunk 20]
```diff
@@ -0,0 +1,407 @@
...
+
+#### `evm_revert`
+
+same as Ganache.
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:339]:
```suggestion
Same as Ganache.
```

File: docs/hardhat-network/reference/README.md [hunk 21]
```diff
@@ -0,0 +1,407 @@
...
+
+Enables (with a numeric argument greater than 0) or disables (with a numeric argument equal to 0), the automatic mining of blocks at a regular interval of milliseconds, each of which will include all pending transactions.
+
+#### `evm_setNextBlockTimestamp`
+
+This method works like `evm_increaseTime`, but takes the exact timestamp that you want in the next block, and increases the time accordingly.
+
+#### `evm_snapshot`
+
+same as Ganache.
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:357]:
```suggestion
Same as Ganache.
```

File: docs/.vuepress/config.js [hunk 22]
```diff
@@ -76,7 +76,16 @@ module.exports = {
...
         ["/getting-started/", "Getting Started", 1],
         ["/config/", "Configuration", 0],
-        ["/hardhat-network/", "Hardhat Network", 0],
+        {
+          title: "Hardhat Network",
+          url: "/hardhat-network/",
+          collapsable: false,
+          depth: 1,
+          children: [
+            ["/hardhat-network/", "Explanation", 0],
```
Comment (alcuadrado) [inline docs/.vuepress/config.js:85]: Maybe something like "What is it?"

File: docs/hardhat-network/README.md (hunk 4 above)
Comment (alcuadrado) [reply docs/hardhat-network/README.md:114]: Mmm, actually, the prioritization in London is much more complex. Txs are sorted by a new concept called "effective miner fee".

Maybe we can use "by the fees payed to the miner", whenever we are talking about sorting/priority?

The second part of the suggestion is valid. You need to increase both EIP-1559 fee params.

File: docs/hardhat-network/reference/README.md [hunk 23]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+This means that no new blocks will be mined by the Hardhat Network, but you can manually mine new blocks using the `evm_mine` RPC method. This will generate a new block that will include as many pending transactions as possible.
+
+### Mempool behavior
+
+When automine is disabled, every sent transaction is added to the mempool, which contains all the transactions that could be mined in the future. Hardhat Network's mempool follows the same rules as geth. This means, among other things, that:
+
+- Transactions with a higher gas price are included first
+- If two transactions can be included and both have the same gas price, the one that was received first is included first
```
Comment (alcuadrado) [inline docs/hardhat-network/reference/README.md:69]: See: https://github.com/nomiclabs/hardhat/pull/1657/files#r674852709

File: docs/hardhat-network/README.md (hunk 1 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/07fecf1d7361787ef2b15f1a3ddface15935791b

File: docs/.vuepress/config.js (hunk 22 above)
Comment (feuGeneA) [reply docs/.vuepress/config.js]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/3383747fecd278d79aa571eb2f06d7fa63997b98

File: docs/hardhat-network/reference/README.md (hunk 17 above)
Comment (feuGeneA) [reply docs/hardhat-network/reference/README.md:289]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/615def3a3ef4e44c89549512b2fc4f916d74fcdb

File: docs/hardhat-network/reference/README.md (hunk 11 above)
Comment (feuGeneA) [reply docs/hardhat-network/reference/README.md:254]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/fc6bc587962d4c7c56e76e7ebc2b0043400bc67c

File: docs/hardhat-network/reference/README.md (hunk 10 above)
Comment (feuGeneA) [reply docs/hardhat-network/reference/README.md:246]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/368a1ff0a7678b23280d5654d9dc67b3fc2d1674

File: docs/hardhat-network/reference/README.md (hunk 7 above)
Comment (feuGeneA) [reply docs/hardhat-network/reference/README.md:15]: @alcuadrado and I had a call about this, and I took these notes on the intended outcome:
```
medium term, give each product its own PAGE, with its own sidebar.
for short term, re-word "Explanation" to "What is it?", add another page to explain mining modes, and link to it in the sidebar as "Mining Modes", and continue the convention of referring to the Divio content types in a markdown comment in the top-level heading of the page, as already done in the hardhat-network/README.md document.
```
I implemented the short term approach here: https://github.com/nomiclabs/hardhat/pull/1657/commits/3383747fecd278d79aa571eb2f06d7fa63997b98

File: docs/hardhat-network/README.md (hunk 5 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md]: I was actually re-reading yesterday and suspecting exactly that.  Thanks for calling this out.  Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/ac503f46f30a8233bbbc13f2af222bff3c9d7101

File: docs/hardhat-network/README.md (hunk 3 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md]: What I actually did was move the details, of what used to be here in this doc, over to the Reference, and left a sort of summary explanation here.  Given your comment here, I just deleted this section: https://github.com/nomiclabs/hardhat/pull/1657/commits/6e4cd0711c19fd62d2d00dd5304684627dfa20ec

File: docs/hardhat-network/README.md (hunk 2 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/c1a6d683c159615b60318bfd3842f086c6740ec8

**Comment by feuGeneA** [comment]
> I think this now conflicts with the changes I made in this PR: #1705

I've manually resolved those conflicts when merging `master` into this branch.

File: docs/hardhat-network/README.md (hunk 4 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/8dbda2c4991fa3a592c288ffa0ae93e60ac21d80

File: docs/hardhat-network/reference/README.md (hunk 23 above)
Comment (feuGeneA) [reply docs/hardhat-network/reference/README.md]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/20aa02e0a31da82939a609c8ccb57c6d38b9401d

File: docs/hardhat-network/reference/README.md (hunk 9 above)
Comment (feuGeneA) [reply docs/hardhat-network/reference/README.md]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/5dee7c7aceb2857be602261237b276c993f50dc9

File: docs/_redirects [hunk 24]
```diff
@@ -339,3 +339,6 @@ https://www.usehardhat.com/* https://hardhat.org/:splat 301!
 # BDLR902
 /BDLR902 /errors/buidler-errors.html#BDLR902 302
 /bdlr902 /errors/buidler-errors.html#BDLR902 302
+
+# Hardhat Network doc consolidation
+/docs/guides/mainnet-forking.md /docs/hardhat-network/guides/mainnet-forking.md
```
Comment (fvictorio) [inline docs/_redirects:344]: Is this working? It doesn't look correct comparing it with the other redirects.

File: docs/hardhat-network/README.md [hunk 25]
```diff
@@ -1,41 +1,44 @@
-# Hardhat Network
+# Hardhat Network <!-- An Explanation: theoretical, not practical; for studying, not for working-->
 
-Hardhat comes built-in with Hardhat Network, a local Ethereum network designed for development. It allows you to deploy your contracts, run your tests and debug your code.
+<!-- TODO: make all RPC method mentions be links to the reference on those methods. -->
+
+Hardhat comes built-in with Hardhat Network, a local Ethereum network node designed for development, akin to Ganache, `geth --devnet`, etc. It allows you to deploy your contracts, run your tests and debug your code.
```
Comment (fvictorio) [inline docs/hardhat-network/README.md:5]:
```suggestion
Hardhat comes built-in with Hardhat Network, a local Ethereum network node designed for development, akin to Ganache, `geth --dev`, etc. It allows you to deploy your contracts, run your tests and debug your code.
```

File: docs/config/README.md [hunk 26]
```diff
@@ -52,56 +52,11 @@ You can customize which network is used by default when running Hardhat by setti
 
 ### Hardhat Network
 
-Hardhat comes built-in with a special network called `hardhat`. When using this network, an instance of the [Hardhat Network](../hardhat-network/README.md) will be automatically created when you run a task, script or test your smart contracts
+Hardhat comes built-in with a special network called `hardhat`. When using this network, an instance of the [Hardhat Network](../hardhat-network/README.md) will be automatically created when you run a task, script or test your smart contracts.
 
 Hardhat Network has first-class support of Solidity. It always knows which smart contracts are being run and exactly what they do and why they fail. Learn more about it [here](../hardhat-network/README.md).
```
Comment (fvictorio) [inline docs/config/README.md:57]: This is not part of this PR, but worth mentioning that this paragraph is misleading. It doesn't always know which smart contracts are being run, nor why they fail. The idea that is a best-effort thing should be made explicit here.

File: docs/hardhat-network/README.md [hunk 27]
```diff
@@ -82,164 +85,37 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
 
 - Trying to deploy a contract that exceeds the bytecode size limit imposed by [EIP-170](https://eips.ethereum.org/EIPS/eip-170)
```
Comment (fvictorio) [inline docs/hardhat-network/README.md:86]: I wonder if we should drop this list, or reduce it to only three or four items that showcase part of the functionality (and maybe move the full list to a more referencish section, assuming this is something useful to have, although my guess is that it doesn't add a lot of value).

File: docs/hardhat-network/README.md [hunk 28]
```diff
@@ -82,164 +85,37 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
...
-To fork from the mainnet you need the URL of a node to connect to. For example, using Alchemy, you can start a local node that forks the mainnet with this command:
+### `console.log`
 
-```
-npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/<key>
-```
+Hardhat Network allows you to print logging messages and contract variables by calling `console.log()` from your Solidity code.
 
-where you have to replace `<key>` with your Alchemy API key.
+To use it, you simply import `hardhat/console.sol` and call it. It implements the same formatting options that can be found in Node.js' [`console.log`](https://nodejs.org/dist/latest-v12.x/docs/api/console.html#console_console_log_data_args), which in turn uses [`util.format`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_format_format_args). For example: `console.log("Changing owner from %s to %s", currentOwner, newOwner)`.
```
Comment (fvictorio) [inline docs/hardhat-network/README.md:92]: I think we should explain here that this only works on hardhat (even if it can be inferred from the context), but that contracts will continue to work on mainnet (but that having console.log statements will use a bit of extra gas unnecessarily). This could be done in a notice/aside block.

File: docs/hardhat-network/README.md [hunk 29]
```diff
@@ -82,164 +85,37 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
...
-  }
-}
-```
+### Mainnet forking
 
-This means that if you execute a task that uses the Hardhat Network, that task will start a forked node and run on it.
+Hardhat Network has the ability to copy the state of the mainnet blockchain into your local environment, including all balances and deployed contracts. This is known as "forking mainnet."
 
-There are other things you can do with a forked Hardhat Network, check [our guide](../guides/mainnet-forking.md) to learn more.
+In a local environment forked from mainnet, you can execute transactions to invoke mainnet-deployed contracts, or interact with the network in any other way that you would with mainnet. In addition, you can do anything that supported by a non-forked Hardhat Network: see console logs, get stack traces, or use the default accounts to deploy new contracts.
```
Comment (fvictorio) [inline docs/hardhat-network/README.md:102]:
```suggestion
In a local environment forked from mainnet, you can execute transactions to invoke mainnet-deployed contracts, or interact with the network in any other way that you would with mainnet. In addition, you can do anything supported by a non-forked Hardhat Network: see console logs, get stack traces, or use the default accounts to deploy new contracts.
```

**Comment by fvictorio** [comment]
It feels wrong to have the "Customizing Hardhat Network's behavior" section inside the Mainnet Forking guide. I think it should live elsewhere (either its own page, or a section in another page), and the Mainent Forking guide should link to it.

File: docs/_redirects (hunk 24 above)
Comment (feuGeneA) [reply docs/_redirects]: Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/f85b549f44ab067f215e71adcf804d74e43b8340

File: docs/hardhat-network/README.md (hunk 28 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md:92]: The next paragraph does explain that it only works on Hardhat (and Tenderly, per Pato), and that it's a no-op everywhere else.  I added a bit of extra text there about the gas: https://github.com/nomiclabs/hardhat/pull/1657/commits/170884778b0bb7eb261c8e4480c24296717eba23

**Comment by feuGeneA** [comment]
> It feels wrong to have the "Customizing Hardhat Network's behavior" section inside the Mainnet Forking guide. I think it should live elsewhere (either its own page, or a section in another page), and the Mainent Forking guide should link to it.

Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/9a62af1ea8bbe2da3b341c5cdecbea987764d326

File: docs/hardhat-network/README.md (hunk 27 above)
Comment (feuGeneA) [reply docs/hardhat-network/README.md:86]: I kinda like it here.  I think you're right that it doesn't add enough value to be added to the reference or anywhere else, but it *does* add value here in that it shows how many situations are covered, helping to drive home the answer to "Why would I want to use it?"  While it may feel nicely succinct to only show a few examples, I think it's easy enough for people to just scroll through it as it is right now.

File: docs/config/README.md [hunk 30]
```diff
@@ -52,56 +52,11 @@ You can customize which network is used by default when running Hardhat by setti
...
-  - `blockNumber`: an optional number to pin which block to fork from. If no value is provided, the latest block is used.
-  - `enabled`: an optional boolean to switch on or off the fork functionality. Default value: `true` if `url` is set, `false` otherwise.
-
-- `minGasPrice`: The minimum `gasPrice` that a transaction must have. This field must not be present if the `"hardfork"` is `"london"` or a later one. Default value: `"0"`.
-
-- `initialBaseFeePerGas`: The `baseFeePerGas` of the first block. Note that when forking a remote network, the "first block" is the one immediately after the block you forked from. This field must not be present if the `"hardfork"` is not `"london"` or a later one. Default value: `"1000000000"` if not forking. When forking a remote network, if the remote network uses EIP-1559, the first local block will use the right `baseFeePerGas` according to the EIP, otherwise `"10000000000"` is used.
+See [the Hardhat Network Configuration Reference](../hardhat-network/reference/#config) for details on what can be configured.
```
Comment (alcuadrado) [inline docs/config/README.md:59]: Sorry to raise this so late, but does this change make sense? 

I think I'd rather have a single page with the entire config, where I can search for values, instead of having to jump between many of them.

**Review by alcuadrado — APPROVED** [review]
This looks good to me now. I just left a minor comment regarding moving the config, but I don't have a strong opinion there either.
